### ISSUE TITLE
Build errors on newest git release

### ISSUE THREAD

**Issue by zyklus** [c4milo/fusejs#15]
@thejinx0r -- I realize this isn't released, but since you're the only one actively contributing to this project, I'm hoping you can help.  I get a slew of build errors with your most recent git push, all about template name mis-matches:

```
 CXX(target) Release/obj.target/fusejs/src/bindings.o
../src/bindings.cc:7:5: error: no template named 'NanPersistent'; did you mean 'Persistent'?
    NanPersistent<FunctionTemplate> Fuse::constructor_template;
    ^~~~~~~~~~~~~
    Persistent
/Users/mark/.node-gyp/2.4.0/deps/v8/include/v8.h:616:45: note: 'Persistent' declared here
  template<class F1, class F2> friend class Persistent;
                                            ^
../src/bindings.cc:10:12: error: no template named 'NanPersistent'; did you mean 'Persistent'?
    static NanPersistent<String> filesystem_sym;
           ^~~~~~~~~~~~~
           Persistent
/Users/mark/.node-gyp/2.4.0/deps/v8/include/v8.h:616:45: note: 'Persistent' declared here
  template<class F1, class F2> friend class Persistent;
                                            ^
../src/bindings.cc:11:12: error: no template named 'NanPersistent'; did you mean 'Persistent'?
    static NanPersistent<String> options_sym;
           ^~~~~~~~~~~~~
           Persistent
```

... another 20 errors

**Comment by thejinx0r** [comment]
It's because you are using nan 2.0 for some reason.

I'll have it sorted out soon for nan 2.0.
I also pushed some incompete nan 2.0 code in the last push, but the latest code that i'm using for my filesystem is this commit thejinx0r@48ee41ec4f998d40f47cbb7f99c37031eb8ee183

**Comment by zyklus** [comment]
> It's because you are using nan 2.0 for some reason.

Yes, io.js and the v0.12 branch both use it I think.

Thanks, that commit builds.  How complete is it, however?  03f9768 (the commit after that one) seems to be fairly major -- is that a re-write or adding new functionality?

**Comment by thejinx0r** [comment]
I was mostly implementing the missing features in commit 03f9768. I was getting ready to push the code to this repo here, and then I saw nan 2.0 with iojs 3.0 looming (which has now been released)

The last couple of commits that I just pushed (4d27c69f904853398c19e7139fa0b80969f9e8f1 and bug fix 9d6887df0d7bd359982339a0b05e8110150a7ac7) are major rewrites for nan 2.0 and is working for the most part for me. I haven't tested fully yet, but the basics seem to be working.
